include/runtime/collections/rangemap.h

```diff
@@ -245,6 +265,41 @@ class RangeMap {
     }
   }
 
+  // Gather the intersection ranges of this rangemap and rangemap m, i.e. all
+  // ranges in this rangemap that overlap (fully or partially) with ranges in m
+  // and are mapped to the same value in both rangemaps, in res.
+  void
+  get_intersection_ranges(RangeMap const &m, std::vector<Range<T>> &res) const {
```

Comment:
There might be usages outwith the files in this diff, but I only see the method being called as:
```cpp
vector<...> v;
get_intersection_ranges(m, v);
```

Could it be refactored to just return the result? If it's called somewhere such that that wouldn't work then no problem.

Reply:
Yes, addressed.